# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1222352357 14400
#      Thu Sep 25 10:19:17 2008 -0400
# Node ID a9ec011ead94f8cf7b92008f934dfd2a2fdb8767
# Parent  7ef5b1b4e0298d15ff49963f4dde716e6ff5b485
make dispatch work with feval

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-09-25  John W. Eaton  <jwe@octave.org>
+
+	* symtab.cc (symbol_table::do_find): Don't set evaluated_args and
+	args_evaluated here, prior to call to symbol_table::fcn_info::find.
+
 2008-09-24  Soren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (magick_to_octave_value): New
 	template function with specializations for various
 	GraphicsMagick++ types.
 	(F__magick_finfo__): New function.
 
 2008-09-24  John W. Eaton  <jwe@octave.org>
diff --git a/src/symtab.cc b/src/symtab.cc
--- a/src/symtab.cc
+++ b/src/symtab.cc
@@ -1067,22 +1067,17 @@ symbol_table::do_find (const std::string
 		return val;
 	    }
 	}
     }
 
   fcn_table_iterator p = fcn_table.find (name);
 
   if (p != fcn_table.end ())
-    {
-      evaluated_args = octave_value_list ();
-      args_evaluated = false;
-
-      return p->second.find (args, arg_names, evaluated_args, args_evaluated);
-    }
+    return p->second.find (args, arg_names, evaluated_args, args_evaluated);
   else
     {
       fcn_info finfo (name);
 
       octave_value fcn = finfo.find (args, arg_names, evaluated_args,
 				     args_evaluated);
 
       if (fcn.is_defined ())
